feature_engine/mathematical_combination.py

```diff
@@ -0,0 +1,94 @@
+from feature_engine.base_transformers import BaseNumericalTransformer
+
+
+class MathematicalVariableCombinator(BaseNumericalTransformer):
+    """
+    The MathematicalVariableCombinator() applies basic mathematical operations across features,
+    returning one additional feature as a result.
+
+    Parameters
+    ----------
+
+    variables: list, default=None
+        The list of numerical variables to be transformed. If None, the transformer
+        will find and select all numerical variables.
+
+    math_operations: list, default=['sum', 'prod', 'mean', 'std', 'max', 'min']
+        The list of basic math operations to be used in transformation.
+
+        Each operation should be a string and must be one of elements
+        from list: ['sum', 'prod', 'mean', 'std', 'max', 'min']
+
+        Each operation will result in operation column in result dataset.
+
+    variables_set_alias: string, default=None
+        Alias for variables set, allow for replacing all columns name from result column with this string.
+
+        If math_operation list contain more than one element, operation will be added to variables_set_alias.
+    """
+
+    def __init__(self, variables=None, math_operations=['sum', 'prod', 'mean', 'std', 'max', 'min'],
+                 variables_set_alias=None):
+        self.variables = variables
+        self.variables_set_alias = variables_set_alias
+        _math_operations_permitted = ['sum', 'prod', 'mean', 'std', 'max', 'min']
+
+        if isinstance(math_operations, list):
+            if any(elem_par not in _math_operations_permitted for elem_par in math_operations):
+                raise KeyError("At least one of math_operations is not found in permitted operations set. "
+                               "Choose one of ['sum', 'prod', 'mean', 'std', 'max', 'min']")
+            self.operations = math_operations
+        else:
+            raise KeyError("math_operations parameter must be a list or None")
+
+        if self.variables and len(self.variables) <= 1:
+            raise KeyError(
+                "MathematicalVariableCombinator requires two or more features to make proper transformations.")
+
+    def fit(self, X, y=None):
+        """
+        Fits source dataset, verifies if variables parameter contains more than one variable.
+
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The training input samples.
+            Can be the entire dataframe, not just the variables to transform.
+
+        y : None
+            y is not needed in this transformer. You can pass y or None.
+        """
+        X = super().fit(X, y)
+        self.input_shape_ = X.shape
```

Comment:
Should we add a dictionary attribute: combination_dict_, that contains the mappings of the new column names to the transformation applied to the features? for example: combination_dict_ = {'total_payments':'sum', 'mean_payments':'mean'}

What do you think?

I am thinking, what if 3 months later you want to know what this transformer did, maybe this helps? thoughts? with this dictionary plus the variables attribute, I think the user should be in a good shape to reconstruct this.? maybe it is an overkill?

Reply:
I think it's a very nice idea. What's more, we will use this dictionary in `transform` method to replace this strange "if-then" code.